### PR TITLE
Picture set refactoring

### PR DESCRIPTION
**Pull request by digitaltom**

### CODE REVIEW

File: app/models/picture_set.rb [hunk 1]
```diff
@@ -6,63 +6,44 @@ class PictureSet
...
+
   class << self
 
     def all
       dirs = Dir.glob(File.join(PICTURE_PATH, "*/*#{ANIMATION_SUFFIX}")).map do |animation|
         File.basename(File.dirname(animation))
       end
-      dirs.sort.reverse.map { |dir| new(dir) }
+      Rails.logger.warn "No picture sets found at: #{PICTURE_PATH}" if dirs.empty?
+      dirs.sort.reverse.map { |dir| PictureSet.new(date: dir) }
```
Comment (ChrisBr) [inline app/models/picture_set.rb]: ``new`` should be enough then you also don't have the dependency to the class name.

File: app/models/picture_set.rb [hunk 2]
```diff
@@ -85,13 +66,59 @@ def capture_job(num, date, dir, angle)
...
+      Rails.logger.info "Skipping for existing animation #{dir}"
+    else
+      Rails.logger.info "Creating animation for #{dir}"
+      Syscall.execute("time convert -delay 60 #{date}_*#{POLAROID_SUFFIX} #{animation}", dir: dir)
+    end
+  end
+
+  # Create collage of all images
+  def combine_images(overwrite: false)
+    if File.exist?(File.join(dir, combined)) && !overwrite
```
Comment (ChrisBr) [inline app/models/picture_set.rb]: Maybe you can also introduce a flag for ``create_combine_images`` which we can set in the options. Then we don't need to comment it on the raspberry ...

File: app/models/picture_set.rb [hunk 3]
```diff
@@ -85,13 +66,59 @@ def capture_job(num, date, dir, angle)
...
+    @date = date
+    @path = "picture_sets/#{date}"
+    @dir = "#{PICTURE_PATH}/#{date}"
+    @animation = "#{date}#{ANIMATION_SUFFIX}"
+    @combined = "#{date}#{COMBINED_SUFFIX}"
+    @pictures = (1..4).map { |i| { polaroid: "#{date}_#{i}#{POLAROID_SUFFIX}", full: "#{date}_#{i}.jpg" } }
+  end
+
+  def destroy
+    Syscall.execute("rm -r #{date}", dir: PICTURE_PATH)
```
Comment (ChrisBr) [inline app/models/picture_set.rb]: Any reason why you don't use FileUtils.rm_rf?
https://ruby-doc.org/stdlib-1.9.3/libdoc/fileutils/rdoc/FileUtils.html#method-c-rm_f

File: app/models/picture_set.rb (hunk 2 above)
Comment (digitaltom) [reply app/models/picture_set.rb]: -> https://github.com/digitaltom/photobooth/issues/19

File: app/models/picture_set.rb (hunk 3 above)
Comment (digitaltom) [reply app/models/picture_set.rb]: should move all file operations to FileUtils, will do.

**Pull request merged**